syntax = "proto3";

package crypto_condor;

message MldsaTest{
	// The test ID, unique in its set of vectors.
	int32 id = 1;
	// The type of test. One of: valid, invalid, acceptable.
	string type = 2;
	// A comment on the test.
	string comment = 3;
	// Flags that categorize this test.
	repeated string flags = 4;

	// The message to sign.
	bytes msg = 5;
	// The public key.
	bytes pk = 6;
	// The secret key.
	bytes sk = 7;
	// The signature.
	bytes sig = 8;
	// The context string.
	bytes ctx = 9;
}

// A set of ML-DSA test vectors.
message MldsaVectors{
	// The source of the test vectors.
	string source = 1;
	// Description of the source.
	string source_desc = 2;
	// The URL of the source.
	string source_url = 3;
	// Whether these are compliance test vectors or not.
	bool compliance = 4;
	// A dictionary of test flags and their description.
	map<string, string> notes = 5;

	// The ML-DSA parameter set.
	string paramset = 6;
	// The test vectors.
	repeated MldsaTest tests = 7;
}
